### PR TITLE
BaSP-TG-132: Firebase Implementation

### PR DESCRIPTION
**Pull request by GonzaloAsencio**
-Install firebase.
-Create a firebase folder inside helpers.
-In firebase folder create an index.js file and apply firebase logic and configuration.
-Create a file name “PrivateRoutes.js“ in routes folder.
-Apply the private routes in the index.js file inside the routes folder.

-Create an auth folder with the following scripts:

**actions.js:** implement the following actions LOGIN,SIGN-UP,LOGOUT.
**constants.js:** create the following constats LOGIN,SIGN-UP,LOGOUT.
**reducer.js:** create the following reducers LOGIN,SIGN-UP,LOGOUT.
**thunks.js:** create the logic for login and logout.

### CODE REVIEW

**Review by fede230298 — APPROVED** [review]

File: src/Helpers/firebase/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          store.dispatch(
+            loginSuccess({
+              role,
+              email
+            })
+          );
+          sessionStorage.setItem('token', token);
+        }
+      } catch (error) {
+        console.log('Error', error);
```
Comment (ProDIGGY-vinilos) [inline src/Helpers/firebase/index.js:36]: delete this plis

File: src/Routes/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  useEffect(() => {
+    tokenListener();
+  }, []);
+  return (
+    <BrowserRouter>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Switch>
+          <PrivateRoute path="/employee" role="EMPLOYEE" component={EmployeeLayout} />
+          <PrivateRoute path="/admin" role="ADMIN" component={AdminLayout} />
+          {/* <PrivateRoute path="/admin" role="SUPER_ADMIN" component={AdminLayout} /> */}
```
Comment (JuliMonte5) [inline src/Routes/index.js:21]: delete comment

File: src/Helpers/firebase/index.js (hunk 1 above)
Comment (ProDIGGY-vinilos) [inline src/Helpers/firebase/index.js:36]: make this as alert

File: src/redux/auth/constants.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const LOGIN_PENDING = 'LOGIN_PENDING';
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
+export const LOGIN_ERROR = 'LOGIN_ERROR';
+export const SIGN_UP_PENDING = 'SING_UP_PENDING';
+export const SIGN_UP_SUCCESS = 'SING_UP_SUCCESS';
+export const SIGN_UP_ERROR = 'SING_UP_ERROR';
```
Comment (ProDIGGY-vinilos) [inline src/redux/auth/constants.js:6]: change this to SIGN

Comment (MaicolFreire001) [reply src/redux/auth/constants.js]: Done

File: src/Routes/index.js (hunk 2 above)
Comment (MaicolFreire001) [reply src/Routes/index.js]: Done

File: src/Helpers/firebase/index.js (hunk 1 above)
Comment (MaicolFreire001) [reply src/Helpers/firebase/index.js]: Done

**Review by JuliMonte5 — APPROVED** [review]

**Review by ProDIGGY-vinilos — APPROVED** [review]

**Pull request merged**